### PR TITLE
[SYCLomatic] Only try to match AST node in analysis scope

### PR DESCRIPTION
**Pull request by ziranzha**
Signed-off-by: Ziran Zhang <ziran.zhang@intel.com>

### CODE REVIEW

File: clang/include/clang/ASTMatchers/ASTMatchFinder.h [hunk 1]
```diff
@@ -199,6 +199,11 @@ class MatchFinder {
   /// Finds all matches in the given AST.
   void matchAST(ASTContext &Context);
 
+#ifdef SYCLomatic_CUSTOMIZATION
+  /// Finds all matches in the given Decl;
+  void traverseDecl(Decl *, ASTContext &);
```
Comment (tomflinda) [inline clang/include/clang/ASTMatchers/ASTMatchFinder.h]: The parameter declaration style is different with the rest ones in this file, pls keep the code style consistent in the same file.

File: clang/lib/ASTMatchers/ASTMatchFinder.cpp [hunk 2]
```diff
@@ -506,6 +506,13 @@ class MatchASTVisitor : public RecursiveASTVisitor<MatchASTVisitor>,
   bool TraverseTemplateArgumentLoc(TemplateArgumentLoc TAL);
   bool TraverseAttr(Attr *AttrNode);
 
+#ifdef SYCLomatic_CUSTOMIZATION
+    bool TraverseCXXDefaultArgExpr(CXXDefaultArgExpr*,
```
Comment (tomflinda) [inline clang/lib/ASTMatchers/ASTMatchFinder.cpp]: Same with above.

File: clang/lib/DPCT/ASTTraversal.h [hunk 3]
```diff
@@ -177,30 +180,8 @@ class MigrationRule : public ASTTraversal {
...
-  }
-
-  // Check if the node's host file is in the InRoot path.
-  inline bool isInAnalysisScope(SourceLocation &&SL) {
-    return DpctGlobalInfo::isInAnalysisScope(SL);
-  }
-
   // Check if the location has been replaced by the same rule.
-  bool isReplaced(SourceRange &SR) {
+  bool isReplaced(SourceRange SR) {
```
Comment (tomflinda) [inline clang/lib/DPCT/ASTTraversal.h:184]: Why remove reference & from SR?

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**